birdie/src-tauri/src/main.rs

```
@@ -64,168 +70,242 @@ fn force_window_to_front(window: Window) {
+                        Err(e) => {
+                            error!("Failed to open repo config file: {:?}", e);
+                            std::process::exit(1);
+                        }
+                    };
+
+                    match serde_yaml::from_reader(file) {
+                        Ok(repo_config) => Some(repo_config),
+                        Err(e) => {
+                            error!("Failed to deserialize repo config: {:?}", e);
+                            std::process::exit(1);
+                        }
+                    }
+                } else {
+                    None
+                }
             }
+            false => None,
         };
 
-    match utils::process::check_for_process(APP_NAME) {
-        Ok(_) => {}
-        Err(e) => {
-            error!("Failed to check for existing process: {:?}", e);
-            std::process::exit(1);
-        }
-    }
+        let (log_path, _otel_reload_handle) = match repo_config {
+            Some(repo_config) => {
+                let init_result = tauri::async_runtime::block_on(async {
+                    // get username from git config
+                    let (tx, rx) = std::sync::mpsc::channel::<String>();
+                    let git_client = Git::new(PathBuf::from(&config.repo_path), tx);
+
+                    // spin up a thread to log git messages
+                    tauri::async_runtime::spawn(async move {
+                        while let Ok(msg) = rx.recv() {
+                            info!("{}", msg);
+                        }
+                    });
+
+                    let username = git_client.get_username().await?;
+                    logging::init(
+                        VERSION,
+                        APP_NAME,
+                        VERSION,
+                        Some(username),
+                        repo_config.otlp_endpoint.clone(),
+                        repo_config.otlp_headers.clone(),
```

If `otlp_endpoint` or `otlp_headers` are empty strings, maybe we return an error before getting here?

---

the internal logging init function does check for empty, does that fit the bill?

---

just pushed up